src/Client/PullPayment.php

```diff
@@ -0,0 +1,336 @@
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /*
+    |--------------------------------------------------------------------------
+    | Public Pull Payment Methods.
+    |--------------------------------------------------------------------------
+    |
+    | For public interaction with their payouts.
+    |
+    */
+
+    public function getPullPayment(
+        string $pullPaymentId
+    ) {
+        $url = $this->getApiUrl() . 'pull-payments/' .
+                    urlencode($pullPaymentId);
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getPayouts(
+        string $pullPaymentId,
+        bool $includeCancelled
+    ) {
+        $url = $this->getApiUrl() . 'pull-payments/' .
+                    urlencode($pullPaymentId) . '/payouts';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $body = json_encode(
+            [
+                'includeCancelled' => $includeCancelled,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function createPayout(
+        string $pullPaymentId,
+        string $destination,
+        string $amount,
+        string $paymentMethod
+    ) {
+        $url = $this->getApiUrl() . 'pull-payments/' .
+                     urlencode($pullPaymentId) . '/payouts';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'destination' => $destination,
+                'amount' => $amount,
+                'paymentMethod' => $paymentMethod,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getPayout(
+        string $pullPaymentId,
+        string $payoutId
+    ) {
```

Comment:
please add return type hint here of the Result class here e.g. Result\PullPaymentPayout